src/nvim/api/extmark.c

```diff
@@ -366,12 +366,10 @@ Array nvim_buf_get_extmarks(Buffer buffer, Integer ns_id, Object start, Object e
 
 /// Creates or updates an extmark.
 ///
-/// To create a new extmark, pass id=0. The extmark id will be returned.
-/// To move an existing mark, pass its id.
-///
-/// It is also allowed to create a new mark by passing in a previously unused
-/// id, but the caller must then keep track of existing and unused ids itself.
-/// (Useful over RPC, to avoid waiting for the return value.)
+/// By default a new extmark is created when no id is passed in, but it is also
```

Comment:
the sentence is awkward with multiple "but" transitions

```suggestion
/// By default a new extmark is created when no id is passed in. It is also
```

Reply:
Fixed.